src/extractor/CaretValueRuleExtractor.ts

```diff
@@ -112,38 +112,65 @@ export class CaretValueRuleExtractor {
       if (flatParent[key] == null || !isEqual(flatSD[key], flatParent[key])) {
         const caretValueRule = new ExportableCaretValueRule('');
         caretValueRule.caretPath = key;
-        caretValueRule.value = getFSHValue(
-          key,
-          flatSD[key],
-          fisher.fishForFHIR('StructureDefinition', utils.Type.Resource),
-          fisher
-        );
+        caretValueRule.value = getFSHValue(key, flatSD[key], 'StructureDefinition', fisher);
         caretValueRules.push(caretValueRule);
       }
     });
 
     return caretValueRules;
   }
+
+  static processResource(
+    input: any,
+    fisher: utils.Fishable,
+    resourceType: 'ValueSet' | 'CodeSystem'
+  ): ExportableCaretValueRule[] {
+    const caretValueRules: ExportableCaretValueRule[] = [];
+    const flatVS = getPathValuePairs(input);
+    Object.keys(flatVS)
+      .filter(
+        key => !RESOURCE_IGNORED_PROPERTIES[resourceType].some(property => key.startsWith(property))
```

Comment:
I don't think that `key.startsWith(property)` is strict enough.  For example, wouldn't the key `identifier` match on the property `id` (and then get unintentionally ignored)?

I think we ought to be checking ``key === property || key.startsWith(`${property}.`)``, right?

Reply:
Yeah good catch. It was actually a little more complex, since in the second case, we would also want to ignore keys like `compose.include[0].system`, but I added a regex that I think should do this correctly.

Reply:
Oooh, yeah.  Good thinking.